code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -6,31 +6,53 @@ What are the archived variables for?
 #define MINIMUM_HEAT_CAPACITY	0.0003
 #define QUANTIZE(variable)		(round(variable,0.0000001))/*I feel the need to document what happens here. Basically this is used to catch most rounding errors, however it's previous value made it so that
 															once gases got hot enough, most procedures wouldnt occur due to the fact that the mole counts would get rounded away. Thus, we lowered it a few orders of magnititude */
-var/list/meta_gas_info = meta_gas_list() //see ATMOSPHERICS/gas_types.dm
-
-var/list/gaslist_cache = null
-/proc/gaslist(id)
-	var/list/cached_gas
-
-	//only instantiate the first time it's needed
-	if(!gaslist_cache)
-		gaslist_cache = new(meta_gas_info.len)
-
-	//only setup the individual lists the first time they're needed
-	if(!gaslist_cache[id])
-		if(!meta_gas_info[id])
-			CRASH("Gas [id] does not exist!")
-		cached_gas = new(3)
-		gaslist_cache[id] = cached_gas
-
-		cached_gas[MOLES] = 0
-		cached_gas[ARCHIVE] = 0
-		cached_gas[GAS_META] = meta_gas_info[id]
-	else
-		cached_gas = gaslist_cache[id]
-	//Copy() it because only GAS_META is static
+#define INIT_GASES new(gaslist_cache.len)	//To be used to setup an empty gases var
+
+var/list/meta_gas_info	//contains the static information of a gas type
+
+var/list/gaslist_cache	//pre initialized list of gases keyed by id, ONLY FOR COPYING
+
+var/list/null_gases	//for sanic loops

```

Comment:
world already has an init() proc.

you should be using factories on these, not defining them on first use.

You aren't saving any overhead, and actually creating overhead by having to make sure these lists exist.

In the land of atmos, even a unneeded if() is a sin.